drivers/overlay/ov_network.go

```diff
@@ -1068,15 +1068,9 @@ func (n *network) contains(ip net.IP) bool {
 }
 
 // getSubnetforIP returns the subnet to which the given IP belongs
-func (n *network) getSubnetforIP(ip *net.IPNet) *subnet {
+func (n *network) getSubnetforIP(ip net.IP) *subnet {
 	for _, s := range n.subnets {
-		// first check if the mask lengths are the same
-		i, _ := s.subnetIP.Mask.Size()
-		j, _ := ip.Mask.Size()
-		if i != j {
-			continue
-		}
-		if s.subnetIP.Contains(ip.IP) {
+		if s.subnetIP.Contains(ip) {
```

Comment:
@mavenugo can you confirm me that this is safe. Networks are never overlapping so the check on the netmask should not be needed correct?